core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -254,40 +251,35 @@ public void start(WorkerNetAddress address) throws IOException {
    * Stops the block worker. This method should only be called to terminate the worker.
    */
   @Override
-  public void stop() {
+  public void stop() throws IOException {
     // Steps to shutdown:
     // 1. Gracefully shut down the runnables running in the executors.
-    // 2. Shutdown the executors.
-    // 3. Shutdown the clients. This needs to happen after the executors is shutdown because
+    // 2. Shutdown the clients. This needs to happen after the executors is shutdown because
     //    runnables running in the executors might be using the clients.
+    // 3. Shutdown base worker. (closes executors.)
```

Comment:
Should we have a mechanism for ensuring the shutdown of resources which need to be closed/stopped(ie. Closer)?

Reply:
https://github.com/Alluxio/alluxio/pull/10777/commits/41ad003489b98ee5d989f77a4d9bc38cb768c45c